gpflow/optimizers/scipy.py

```diff
@@ -31,6 +43,13 @@
 
 
 class Scipy:
+    def __init__(self) -> None:
```

Comment:
How about making `compile_cache_limit: int = 2` an initialistion argument? That would allow both changing the cache size and (more importantly) disabling it (by setting it to 0) if it turns out there is an issue that we haven't foreseen.

Reply:
I have added this now.